python/activator/middleware_interface.py

```diff
@@ -40,6 +40,20 @@
 
 
 class MiddlewareInterface:
+    """Interface layer between the Butler middleware and the prompt processing
+    data handling system.
+
+    Parameters
```

Comment:
Can you include an explanation of what a `MiddlewareInterface` object actually represents? I find it very confusing that `__init__` performs Butler operations, some of them ones that change state.

The fact that some (but not all) of the code assumes multiple concurrent `MiddlewareInterface` objects is probably a clue...

Reply:
I've added a paragraph, but I think we don't want to assume that any of the stuff in `__init__` will stick around (I suspect most of that should move to `prep_butler`, but don't know yet).

Reply:
Of course, and we shouldn't document implementations anyway. I just meant that the class seems to be intended to be used in particular ways, which are not at all obvious from the name.